python/GafferSceneUI/MeshTangentsUI.py

```diff
@@ -36,51 +36,102 @@
 
 import Gaffer
 import GafferScene
+import IECoreScene
 
 Gaffer.Metadata.registerNode(
 
 	GafferScene.MeshTangents,
 
 	"description",
 	"""
-	Adds surface tangent primitive variables to the mesh using the specified positions and UVSet. 
+	Adds surface tangent primitive variables to the mesh using the specified positions and UVSet.
 	""",
 
+	"layout:activator:uvActivator", lambda parent : parent["mode"].getValue() == int(IECoreScene.MeshAlgo.MeshTangentsMode.UV),
+	"layout:activator:leftHandedActivator", lambda parent : parent["orthogonal"].getValue() == True,
+
 	plugs = {
 
-		"uvSet" : [
+		"mode" : [
+
 			"description",
 			"""
-			Name of the UV set primitive variable used to calculate uTangent & vTangent.
+			The style of how to calculate the Tangents.
+			(UV) calculates the tangents based on the gradient of the the corresponding UVs
+			(FirstEdge) defines the vector to the first neighbor as tangent and the bitangent orthogonal to tangent and normal
+			(TwoEdges) defines the vector between the first two neighbors as tangent and the bitangent orthogonal to tangent and normal
+			(PrimitiveCentroid) points the tangent towards the primitive centroid and the bitangent orthogonal to tangent and normal
 			""",
+
+			"preset:UV", IECoreScene.MeshAlgo.MeshTangentsMode.UV,
+			"preset:FirstEdge", IECoreScene.MeshAlgo.MeshTangentsMode.FirstEdge,
+			"preset:TwoEdges", IECoreScene.MeshAlgo.MeshTangentsMode.TwoEdges,
+			"preset:PrimitiveCentroid",IECoreScene.MeshAlgo.MeshTangentsMode.PrimitiveCentroid,
+
+			"plugValueWidget:type", "GafferUI.PresetsPlugValueWidget",
+
+			"layout:index", 4,
```

Comment:
It's a bit fragile specifying the order this way (despite being necessary sometimes), and it isn't respected when we auto-generate the docs. Let's just construct the plugs in the order we want, and order the accessors in the header correspondingly. That way we're presenting things in the same logical order wherever you look.